lib/resourceapply/security_test.go

```diff
@@ -0,0 +1,122 @@
+package resourceapply
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	securityv1 "github.com/openshift/api/security/v1"
+	securityfake "github.com/openshift/client-go/security/clientset/versioned/fake"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var restrictedv2 = securityv1.SecurityContextConstraints{
+	TypeMeta: metav1.TypeMeta{Kind: "SecurityContextConstraints", APIVersion: securityv1.GroupVersion.String()},
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "restricted-v2",
+		Annotations: map[string]string{
+			"include.release.openshift.io/ibm-cloud-managed":              "true",
+			"include.release.openshift.io/self-managed-high-availability": "true",
+			"include.release.openshift.io/single-node-developer":          "true",
+			"kubernetes.io/description":                                   "restricted-v2 denies access...",
+		},
+	},
+	RequiredDropCapabilities: []corev1.Capability{"ALL"},
+	AllowedCapabilities:      []corev1.Capability{"NET_BIND_SERVICE"},
+	Volumes:                  []securityv1.FSType{"configMap", "downwardAPI", "emptyDir", "persistentVolumeClaim", "projected", "secret"},
+	SELinuxContext:           securityv1.SELinuxContextStrategyOptions{Type: securityv1.SELinuxStrategyMustRunAs},
+	RunAsUser:                securityv1.RunAsUserStrategyOptions{Type: securityv1.RunAsUserStrategyMustRunAsRange},
+	SupplementalGroups:       securityv1.SupplementalGroupsStrategyOptions{Type: securityv1.SupplementalGroupsStrategyRunAsAny},
+	FSGroup:                  securityv1.FSGroupStrategyOptions{Type: securityv1.FSGroupStrategyMustRunAs},
+	SeccompProfiles:          []string{"runtime/default"},
+}
+
+func TestApplySecurityContextConstraintsv1(t *testing.T) {
```

Comment:
Add tests for scenarios with "create-only" annotation on the SCC. If you accidentally reconciled those, very bad things would start to happen.

Reply:
added a test